src/react/DevicesPage.tsx

```diff
@@ -50,7 +66,111 @@ const matchMetadata = (key, value, filterKey, filterValue) => {
   return true;
 };
 
-const matchFilters = (device, filters) => {
+interface MatchedMetadataProps {
+  filters: DeviceFilters;
+  metadata: AstarteDevice['metadata'];
+}
+
+const MatchedMetadata = ({
+  filters,
+  metadata,
+}: MatchedMetadataProps): React.ReactElement | null => {
```

Comment:
if you return a `React.ReactElement[]` you can replace `null -> []` and remove the wrapping fragment before the return

Reply:
If I did that, Typescript would complain that a React component needs to return a single root element and that you would use fragments to package an array of elements as a single element.